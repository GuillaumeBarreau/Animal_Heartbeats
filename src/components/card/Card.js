import React from 'react';
import { connect } from 'react-redux';
import { convertToIntFcm } from '../../logic/convertToIntFcm.logic';
// Import Styled components
import { Item, Link, Title, ContentHearth, Hearth, Legende, Text, Avatar } from './Card.styled';

const Card = ({ theme, ...props }) => (
	<Item id={props.name} theme={theme}>
		<Link theme={theme} href="">
			<Title>{props.name}</Title>
			{props.image ? <Avatar /> : null}
			<ContentHearth>
				<Hearth fcm={convertToIntFcm(props.fcm)}></Hearth>
			</ContentHearth>
			<Text>â‰ˆ {convertToIntFcm(props.fcm)} BPM</Text>
			<Legende>BPM : Beats per minute</Legende>
		</Link>
	</Item>
);

const mapStateToProps = state => {
	return {
		theme: state.theme,
	};
};

export default connect(mapStateToProps)(Card);
