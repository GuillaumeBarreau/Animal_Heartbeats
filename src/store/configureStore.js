import { createStore, applyMiddleware } from 'redux';
import reducers from './reducers/combineReducers';

function logger({ getState }) {
	return next => action => {
		console.log('will dispatch', action);
		return next(action);
	};
}

export const store = createStore(reducers, applyMiddleware(logger));
 