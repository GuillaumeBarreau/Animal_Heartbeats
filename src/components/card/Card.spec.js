import { shallow } from 'enzyme';
import React from 'react';
import Card from './Card';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);

describe('Card component', () => {
	let store;
	let props;

	beforeEach(() => {
		store = mockStore({
			theme: 'dark',
		});
		props = {
			name: 'name props',
			fcm: '100',
		};
	});

	it('Render Card component', () => {
		const cmp = shallow(
			<Provider store={store}>
				<Card {...props} />
			</Provider>
		);
		expect(cmp.exists()).toBe(true);
	});

	it('Snapshot testing', () => {
		const cmp = shallow(
			<Provider store={store}>
				<Card {...props} />
			</Provider>
		);
		expect(cmp.html()).toMatchSnapshot();
	});

	it('should display a txt name props', () => {
		const cmp = shallow(
			<Provider store={store}>
				<Card {...props} />
			</Provider>
		);
		expect(cmp.html()).toContain('name props');
	});

	it('should display a txt fcm props', () => {
		const cmp = shallow(
			<Provider store={store}>
				<Card {...props} />
			</Provider>
		);
		expect(cmp.html()).toContain('100');
	});
});
