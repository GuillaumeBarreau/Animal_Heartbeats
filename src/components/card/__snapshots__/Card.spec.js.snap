// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Snapshot testing 1`] = `"<li id=\\"name props\\" class=\\"sc-AykKC gTkJkp\\"><a href=\\"\\" class=\\"sc-AykKD heYvfS\\"><h2 class=\\"sc-AykKE dtaabF\\">name props</h2><div class=\\"sc-AykKF byrLDp\\"><div class=\\"sc-AykKG cpvmWV\\"></div></div><p class=\\"sc-AykKI fSiutg\\">â‰ˆ 100 BPM</p><p class=\\"sc-AykKH bdhmnO\\">BPM : Beats per minute</p></a></li>"`;
